include/mqtt/shared_ptr_array.hpp

```diff
@@ -47,11 +47,17 @@ using shared_ptr_array = std::shared_ptr<char []>;
 using const_shared_ptr_array = std::shared_ptr<char const []>;
 
 inline shared_ptr_array make_shared_ptr_array(std::size_t size) {
-#if __cplusplus > 201703L // C++20 date is not determined yet
+#ifdef __cpp_lib_shared_ptr_arrays
+#if __cpp_lib_shared_ptr_arrays >= 201707L
```

Comment:
```suggestion
#if __cpp_lib_shared_ptr_arrays >= 201707L
```

If `__cpp_lib_shared_ptr_arrays` is not defined, `__cpp_lib_shared_ptr_arrays` is evaluated as 0.

Reply:
Ok, I am never sure about macros :). A different but related feature testing macro requires this separated #ifdef check, see https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005fattribute.html

Reply:
`__has_attribute` is a function style macro, so we need to check its existance.